src/client/src/reducers/index.ts

```diff
@@ -21,5 +23,33 @@ const rootReducer = combineReducers({
   dependencyInfo
 });
 
-export type AppState = ReturnType<typeof rootReducer>;
+export type AppState = ReturnType<typeof appReducer>;
+
+const rootReducer = (state: AppState | undefined, action: RootAction) => {
+  let passedState: any;
+
+  if (action.type === WIZARD_INFO_TYPEKEYS.RESET_WIZARD) {
+    const { previewStatus } = state!.wizardContent;
+
+    /* Elements that are undefined tell the reducer to replace the element
+     * with the initial state that is specified in the element's reducer.
+     * See: https://redux.js.org/recipes/structuring-reducers/initializing-state
+     */
+    passedState = {
+      azureProfileData: state!.azureProfileData,
+      dependencyInfo: undefined,
+      generationStatus: undefined,
+      modals: undefined,
+      selection: undefined,
+      versions: state!.versions,
+      vscode: state!.vscode,
+      wizardContent: { previewStatus },
```

Comment:
Just for confirmation, correct me if I am saying it wrong. We are assigning `wizardContent` to have `previewStatus` property as defined above. All other attributes of `wizardContent` would be `undefined`, so they will take up the initial state from their respective reducers. Is that right?

Reply:
Yup all other attributes will be undefined!